api/nodecontrol/client_test.go

```diff
@@ -0,0 +1,943 @@
+)
+
+func buildTestBinary(t testing.TB, binMain string, workingDir string) (string, error) {
+	t.Helper()
+	binName := func() string {
+		if runtime.GOOS == "windows" {
+			return "test.exe"
+		}
+		return "test"
+	}
+
+	if _, err := os.Stat(filepath.Join(workingDir, binName())); err == nil {
+		return filepath.Join(workingDir, binName()), nil
+	}
+
+	cmd := exec.Command("go", "build", "-o", filepath.Join(workingDir, binName()), binMain)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	err := cmd.Run()
+	if err != nil {
+		return "", err
+	}
+
+	if _, err := os.Stat(filepath.Join(workingDir, binName())); err != nil {
+		return "", err
+	}
+
+	return filepath.Join(workingDir, binName()), nil
+}
+
+func startNatsServer(t testing.TB, workingDir string) (*server.Server, error) {
+	t.Helper()
+
+	s := server.New(&server.Options{
+		Port:      -1,
+		JetStream: true,
+		StoreDir:  workingDir,
+	})
+
+	s.Start()
+
+	go s.WaitForShutdown()
+
+	return s, nil
+}
+
+func startNexus(t testing.TB, ctx context.Context, logger *slog.Logger, workingDir, natsUrl string, numNodes int) error {
+	t.Helper()
+
+	nc, err := nats.Connect(natsUrl)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < numNodes; i++ {
+		var kp, xkp nkeys.KeyPair
+		if i == 0 {
+			kp, err = nkeys.FromSeed([]byte(Node1ServerSeed))
+			if err != nil {
+				return err
+			}
+			xkp, err = nkeys.FromSeed([]byte(Node1XKeySeed))
+			if err != nil {
+				return err
+			}
+		} else {
+			kp, err = nkeys.CreateServer()
+			if err != nil {
+				return err
+			}
+			xkp, err = nkeys.CreateCurveKeys()
+			if err != nil {
+				return err
+			}
+		}
+		nn, err := node.NewNexNode(kp, nc,
+			models.WithContext(ctx),
+			models.WithLogger(logger),
+			models.WithXKeyKeyPair(xkp),
+			models.WithNodeName(fmt.Sprintf("node-%d", i+1)),
+			models.WithNexus("testnexus"),
+			models.WithResourceDirectory(workingDir),
+		)
+		if err != nil {
+			return err
+		}
+
+		err = nn.Validate()
+		if err != nil {
+			return err
+		}
+
+		go func() {
+			err = nn.Start()
+			if err != nil {
+				t.Error(err)
+				t.FailNow()
+			}
+		}()
+	}
+
+	return nil
+}
+
+func TestAuction(t *testing.T) {
+	workingDir := t.TempDir()
+	natsServer, err := startNatsServer(t, workingDir)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+
+	t.Cleanup(func() {
+		os.RemoveAll(filepath.Join(os.TempDir(), "inex-NCUU2YIYXEPGTCDXDKQR7LL5PXDHIDG7SDFLWKE3WY63ZGCZL2HKIAJT"))
+		cancel()
+		natsServer.Shutdown()
+	})
+
+	stdout := new(bytes.Buffer)
+	stderr := new(bytes.Buffer)
+	logger := slog.New(shandler.NewHandler(
+		shandler.WithLogLevel(slog.LevelDebug),
+		shandler.WithGroupFilter([]string{"actor_system"}),
+		shandler.WithStdOut(stdout),
+		shandler.WithStdErr(stderr),
+	))
+
+	err = startNexus(t, ctx, logger, workingDir, natsServer.ClientURL(), 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	time.Sleep(1000 * time.Millisecond)
```

Comment:
I presume this is waiting for the Nexus to start up? I would suggest using a wait group or channel in `startNexus` that blocks until the nodes are up (until some max time) or fail within the function.

Reply:
Yea, when I was writing this, I realized I need a "healthy" check....this will probably be a function of the events subsystem that doesnt exist yet 😞

Reply:
Issue for tracking: https://github.com/synadia-io/nex/issues/475